syntax = "proto3";

package example;

message Person {
  string name = 1;
  int32 age = 2;
  string email = 3;
  repeated string hobbies = 4;
  
  enum Gender {
    UNKNOWN = 0;
    MALE = 1;
    FEMALE = 2;
    OTHER = 3;
  }
  
  Gender gender = 5;
}

message AddressBook {
  repeated Person people = 1;
  string title = 2;
  string created_date = 3;
}

message SimpleMessage {
  string content = 1;
  int64 timestamp = 2;
  bool is_important = 3;
}
